### PR TITLE
[test] Add string testcase to collection

### PR DESCRIPTION
**Pull request by jingkl**
Signed-off-by: jingkl <jingjing.jia@zilliz.com>
issue: #16833

### CODE REVIEW

File: tests/python_client/testcases/test_collection.py [hunk 1]
```diff
@@ -2725,3 +2710,117 @@ def test_load_release_after_collection_drop(self, connect, collection):
...
+
+class TestCollectionString(TestcaseBase):
+    """
+    ******************************************************************
+      The following cases are used to test about string 
+    ******************************************************************
+    """
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_collection_string_field_is_primarty(self):
```
Comment (binbinlv) [inline tests/python_client/testcases/test_collection.py:2723]: test_collection_string_field_is_primarty -> test_collection_string_field_is_primary

Comment (jingkl) [reply tests/python_client/testcases/test_collection.py:2723]: Received, thanks

File: tests/python_client/testcases/test_collection.py [hunk 2]
```diff
@@ -2725,3 +2710,117 @@ def test_load_release_after_collection_drop(self, connect, collection):
...
+        method: 1. create collection with string field and vector field
+                2. set string fields is_primary=True
+        expected: Create  collection Successfully
+        """
+        self._connect()
+        c_name = cf.gen_unique_str(prefix)
+        int_field = cf.gen_int64_field()
+        vec_field = cf.gen_float_vec_field()
+        string_field = cf.gen_string_field(is_primary=True)
+        schema = cf.gen_collection_schema(fields=[int_field, string_field, vec_field])
```
Comment (binbinlv) [inline tests/python_client/testcases/test_collection.py:2735]: why does not use function "gen_default_collection_schema" to replace the above 4 lines for schema generation?

**Review by binbinlv — CHANGES_REQUESTED** [review]

File: tests/python_client/testcases/test_collection.py [hunk 3]
```diff
@@ -2725,3 +2710,117 @@ def test_load_release_after_collection_drop(self, connect, collection):
...
+        self.collection_wrap.init_collection(name=c_name, schema=schema, check_task=CheckTasks.check_collection_property,
+                                             check_items={exp_name: c_name, exp_schema: schema})
+    
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_collection_with_muti_string_fields(self):
+        """
+        target: test create collection with muti string fields
+        method: 1. create  collection with primary string field and not primary string field
+                2.  String fields is_primary=True
+        expected: Create  collection Successfully
```
Comment (binbinlv) [inline tests/python_client/testcases/test_collection.py:2745]: Create  collection Successfully -> Create collection Successfully (two blanks) and the above "string" is not aligned with "create ..." line

File: tests/python_client/testcases/test_collection.py [hunk 4]
```diff
@@ -2725,3 +2710,117 @@ def test_load_release_after_collection_drop(self, connect, collection):
...
+        string_field = self.field_schema_wrap.init_field_schema(name="string", dtype=DataType.STRING)[0]
+        int_field = cf.gen_int64_field(is_primary=True)
+        vec_field = cf.gen_float_vec_field()
+        schema = cf.gen_collection_schema(fields=[int_field, string_field, vec_field])
+        error = {ct.err_code: 0, ct.err_msg: "string data type not supported yet, please use VarChar type instead"}
+        self.collection_wrap.init_collection(name=cf.gen_unique_str(prefix), schema=schema,
+                                             check_task=CheckTasks.err_res, check_items=error)
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_collection_invalid_string_field_with_auto_id(self):
```
Comment (binbinlv) [inline tests/python_client/testcases/test_collection.py:2807]: The test case name did not reflect the functions tested, and the  "test_collection_string_field_is_primary_and_auto_id"  may be better. and the following comment should be modified to reflect what is tested.

File: tests/python_client/testcases/test_collection.py (hunk 2 above)
Comment (jingkl) [reply tests/python_client/testcases/test_collection.py]: Thanks for the suggestion has been modified

File: tests/python_client/testcases/test_collection.py (hunk 4 above)
Comment (jingkl) [reply tests/python_client/testcases/test_collection.py]: Thanks for the suggestion has been modified

**Pull request merged**